polyswarm/operations.py

```diff
@@ -0,0 +1,167 @@
+        logger.info(file_path)
+        return file_path
+        with open(file_path, 'rb') as attachment:
+            file_data = attachment.read()
+        if file_data:
+            files = {'file': file_data}
+            return files
+        raise ConnectorError('File size too large, submit file up to 32 MB')
+    except Exception as Err:
+        logger.error('Error in submitFile(): %s' % Err)
+        raise ConnectorError('Error in submitFile(): %s' % Err)
+
+
+def evaluate_assertions(result):
+    positives = 0
+    total = 0
+    for assertion in result.assertions:
+        if assertion.verdict:
+            positives += 1
+        total += 1
+
+    return {
+        'Positives': positives,
+        'Total': total,
+        'PolyScore': result.polyscore,
+        'sha256': result.sha256,
+        'sha1': result.sha1,
+        'md5': result.md5,
+        'Extended type': result.extended_type,
+        'First Seen': result.first_seen,
+        'Last Seen': result.last_seen,
+        'Permalink': result.permalink
+    }
+
+
+def file_scan(config, params):
+    try:
+        api = api_client_creation(config.get('api_key'), config.get("verify_ssl"))
+        file_iri = handle_params(params)
+        files = submitFile(file_iri)
+        logger.error("file path is {}".format(files))
+        instance = api.submit(files)
+        result = api.wait_for(instance)
+        if result.failed:
+            raise ConnectorError("Failed to get results")
+        return evaluate_assertions(result)
+    except Exception as e:
+        raise ConnectorError(e)
+
+
+def file_rescan(config, params):
+    try:
+        api = api_client_creation(config.get('api_key'), config.get("verify_ssl"))
+        instance = api.rescan(params.get('hash'))
+        result = api.wait_for(instance)
+        if result.failed:
+            raise ConnectorError("Failed to get results")
+        return evaluate_assertions(result)
+    except Exception as e:
+        raise ConnectorError(e)
+
+
+def file_reputation(config, params):
+    try:
+        api = api_client_creation(config.get('api_key'), config.get("verify_ssl"))
+        results = api.search(params.get('hash'))
+        for result in results:
+            if result.failed:
+                raise ConnectorError("Failed to get result.")
+            if not result.assertions:
+                raise ConnectorError('Artifact not scanned yet - Run rescan for this hash')
+            return evaluate_assertions(result)
+    except Exception as e:
+        raise ConnectorError(e)
+
+
+def _check_health(config: dict) -> bool:
+    try:
+        api = api_client_creation(config.get('api_key'), config.get("verify_ssl"))
+        api.submit('https://polyswarm.io', artifact_type='url')
+        return True
+    except Exception as e:
+        raise ConnectorError("Invalid API Key")
+
+
+operations = {
+    'url_reputation': url_reputation,
+    'ip_reputation': url_reputation,
+    'domain_reputation': url_reputation,
```

Comment:
change the function name as it is used for 3 operations, it is not url specific

Reply:
done